### PR TITLE
feat: support ckb v0.22.0

### PR DESCRIPTION
**Pull request by shaojunda**
1. header compress
	* remove uncles_count
	* remove witnesses_root
	* replace difficulty with compact_target
2. change args and witness to string
3. add `compact_to_difficulty` method
4. change `generate_lock_script_from_cellbase` logic

### CODE REVIEW

File: db/schema.rb [hunk 1]
```diff
@@ -165,14 +162,15 @@
     t.datetime "updated_at", null: false
     t.decimal "primary_reward", precision: 30, default: "0"
     t.decimal "secondary_reward", precision: 30, default: "0"
-    t.decimal "difficulty", precision: 80, default: "0"
     t.decimal "nonce", precision: 30, default: "0"
     t.decimal "start_number", precision: 30, default: "0"
     t.decimal "length", precision: 30, default: "0"
+    t.integer "uncles_count"
```
Comment (classicalliu) [inline db/schema.rb]: `uncles_count` haven't removed ?

Comment (shaojunda) [reply db/schema.rb]: yes, although ckb deleted it from header, Explorer still need it. so I change to calculate it manually.

Comment (classicalliu) [reply db/schema.rb]: Ok. but `uncles_count` is an u32, `integer` is an i32

Comment (shaojunda) [reply db/schema.rb]: A normal block should not have too many uncles

**Pull request merged**